templates/geth/geth.py

```diff
@@ -0,0 +1,243 @@
+        }
+        container_data['env'] = self._container_autostart_env
+
+        self._container = self._node_sal.containers.create(**container_data)
+        return self._container
+
+    def install(self):
+        """
+        Creating geth container with the provided flist, and configure mounts for datadirs
+            'flist': GETH_FLIST,
+        """
+        sp = self._node_sal.storagepools.get('zos-cache')
+        try:
+            fs = sp.get(self.guid)
+        except ValueError:
+            fs = sp.create(self.guid)
+
+        self.logger.info('installing geth %s', self.name)
+
+        self.state.set('actions', 'install', 'ok')
+
+    def uninstall(self):
+        """Remove the persistent volume of ethereum, and delete the
+        container."""
+        try:
+            self._container_sal.stop()
+        except Exception as e:
+            self.logger.warning(
+                "removing container on the uninstall {}".format(e))
+        try:
+            # cleanup filesystem used by this robot
+            sp = self._node_sal.storagepools.get('zos-cache')
+            fs = sp.get(self.guid)
+            fs.delete()
+        except ValueError:
+            # filesystem doesn't exist, nothing else to do
+            pass
+
+        self.state.delete('actions', 'install')
+
+    def start(self):
+        """starts geth."""
+        self.state.check('actions', 'install', 'ok')
+
+        self._get_container()
+
+        self.state.set('status', 'running', 'ok')
+        self.state.set('actions', 'start', 'ok')
+
+    def stop(self):
+        """stop geth."""
+        self.logger.info('Stopping geth %s', self.name)
+        self._container_sal.stop()
+
+        self.state.delete('status', 'running')
+        self.state.delete('actions', 'start')
+
+    def upgrade(self, gethFlist=None):
+        """upgrade the container with an updated flist this is done by stopping
+        the container and respawn again with the updated flist.
+
+        gethFlist: If provided, the current used flist will be replaced with the specified one
+        """
+        # update flist
+        if gethFlist:
+            self.data['gethFlist'] = gethFlist
+
+        self.stop()
+        # restart geth in new container
+        self.start()
+
+    def _check_sync(self):
+        """
+        recurring function that updates service state with the ethereum syncing state every 60 seconds. ref init
+        """
+        result = self.getSyncingStatus()
+        if result is None:
+            self.state.delete('ethereum','syncing')
+            self.state.delete('ethereum','synced')
+            return
+
+        currentblock, highestblock = result
+        if currentblock >= highestblock:
+            self.state.set('ethereum','synced', 'ok')
+            self.state.delete('ethereum','syncing')
+        else:
+            self.state.set('ethereum','syncing', 'ok')
+            self.state.delete('ethereum','synced')
+
+    def getSyncingStatus(self):
```

Comment:
function names should be snake cased, `getSycningStatus` -> `get_syncing_status`